src/measure_theory/interval_integral.lean

```diff
@@ -1513,6 +1513,14 @@ theorem integral_deriv_eq_sub' (f) (hderiv : deriv f = f')
   ∫ y in a..b, f' y = f b - f a :=
 by rw [← hderiv, integral_deriv_eq_sub hdiff]; cc
 
+theorem integral_deriv_comp_mul_deriv {f f' g g' : ℝ → ℝ}
```

Comment:
Is that really the right theorem? I mean, it would seem more natural to me to have a statement of the form 
```
∫ x in a..b, (g ∘ f) x * f' x = ∫ x in (f a)..(f b), g x
```
for a general function `f` (not a derivative a priori), and then deduce your statement when `g` is the derivative from the fact that then the right hand side can be computed with FTC.

Reply:
You are correct, and I believe @ADedecker is working on proving a more general version of that very theorem. Once that is added this should be easily derived from there. But this result will remain useful regardless, so I figured we might as well add it now, even if I'll need to re-derive at some later time.

Reply:
I could also add the version you mention now without much difficulty, but I had decided against it since I imagine it will be totally nullified by Anatole's work.

Reply:
I wouldn't say "much more general". Actually, the following proofs have been sitting in my draft folder for about a month now, but I had just forgotten to PR them. I think this is the usual form of 1D variable change, so I don't know if we can make a more general statement.

```lean
lemma variable_change {a b : ℝ} {f φ φ' : ℝ → ℝ} 
  (hφ : ∀ x ∈ interval a b, has_deriv_at φ (φ' x) x)
  (hφ' : continuous_on φ' (interval a b))
  (hf : ∀ x ∈ φ '' (interval a b), continuous_at f x)
  (hfm : ∀ x ∈ φ '' (interval a b), measurable_at_filter f (𝓝 x)) :
  ∫ t in a..b, f (φ t) * (φ' t) = ∫ t in (φ a)..(φ b), f t :=
begin
  have : ∀ x ∈ interval a b, has_deriv_at (λ u, ∫ t in (φ a)..(φ u), f t) (f (φ x) * (φ' x)) x,
  { intros x hx,
    have key₁ : interval a x ⊆ interval a b := interval_subset_interval_left hx,
    have key₂ : interval (φ a) (φ x) ⊆ φ '' interval a x := intermediate_value_interval
      (λ y hy, (hφ y (key₁ hy)).continuous_at.continuous_within_at),
    have key₃ : continuous_on f (φ '' interval a b) := (λ y hy, (hf y hy).continuous_within_at),
    exact (integral_has_deriv_at_right 
      (key₃.mono $ trans key₂ (image_subset φ key₁)).interval_integrable 
      (hfm (φ x) ⟨x, hx, rfl⟩) (hf (φ x) ⟨x, hx, rfl⟩)).comp _ (hφ x hx) },
  have key₄ : continuous_on f (φ '' interval a b) := λ y hy, (hf y hy).continuous_within_at,
  rw integral_eq_sub_of_has_deriv_at this ((key₄.comp 
    (λ x hx, (hφ x hx).continuous_at.continuous_within_at) 
    (subset_preimage_image φ (interval a b))).mul hφ'),
  simp
end

lemma variable_change_of_continuous {a b : ℝ} {f φ φ' : ℝ → ℝ} 
  (hφ : ∀ x ∈ interval a b, has_deriv_at φ (φ' x) x)
  (hφ' : continuous_on φ' (interval a b)) (hf : continuous f) :
  ∫ t in a..b, f (φ t) * (φ' t) = ∫ t in (φ a)..(φ b), f t :=
variable_change hφ hφ' (λ _ _, hf.continuous_at) (λ _ _, hf.measurable.measurable_at_filter)
```

Note : these may not compile with current mathlib, but it should be easily adaptable. 

Feel free to include these into your PR and clean them a little. Otherwise I can PR them in a few days, but I don't care so do what you prefer.

Note 2 : I've been thinking about proving n dimensional variable change, but I've looked at the proof online and I really don't know measure theory enough.

Reply:
Thank you Anatole!! Funny enough, I also had this proof sitting around for a couple of months before I opened this PR last week so I wouldn't;t forget about it again. By "more general" I had meant that you were working on weakening the assumptions, which I see you've managed to do! Your proofs look great and I would be more than happy to adapt them into a pull request. I'll revert this to a WIP while I do that.